Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -144,7 +147,7 @@ public async Task ValidateConnectionStringAsync(string connectionString)
             if (!match.Groups[CONNSTRING_REGEX_KEY].Value.IsNullOrWhiteSpace())
             {
                 this.ValidateConnectionString(connectionString);
-                await this.TestIoTHubReadPermissionsAsync(connectionString);
+                this.TestIoTHubReadPermissionsAsync(connectionString);
                 await this.TestIoTHubWritePermissionsAsync(connectionString);
```

Comment:
Is there a reason why not await on read `this.TestIoTHubReadPermissionsAsync(connectionString);` but await on write at the next line?
                `await this.TestIoTHubWritePermissionsAsync(connectionString);`

Reply:
because the function that replaced the deprecated function is no longer asynchronous and so we don't need to wait on it